src/main/java/wraith/alloyforgery/utils/DataPackEvents.java

```diff
@@ -0,0 +1,33 @@
+package wraith.alloyforgery.utils;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.PlayerManager;
+
+public class DataPackEvents {
+
+    /**
+     * Event Similar to {@link ServerLifecycleEvents#SYNC_DATA_PACK_CONTENTS} with the goal as a
+     * method to adjust Data Pack based data before sync after such has loaded
+     */
+    public static final Event<BeforeSync> BEFORE_SYNC = EventFactory.createArrayBacked(BeforeSync.class, callbacks -> (server) -> {
```

Comment:
I would love a comment explaining why we need to sync before just before resource reload